### PR TITLE
Profile data loads

### PR DESCRIPTION
**Pull request by Timbobeek**
This code fetches data from a profile data api and loads it onto our online page.

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -16,7 +63,12 @@ export default function Home() {
       <main className={styles.main}>
         <div className={styles.center}>
           <div className={styles.description}>
-            <p>Coming soon!</p>
+            Profile Picture URL: {profile?.gitHub?.profile?.avatarUrl}<br/><br/>
+            Name: {profile?.gitHub ? profile?.gitHub?.profile?.name :  "loading..."}<br/><br/>
```
Comment (ferrata) [inline src/pages/index.tsx]: Maybe show "loading..." instead of the whole `div` content?

File: src/pages/index.tsx [hunk 2]
```diff
@@ -1,10 +1,57 @@
...
+  gitHub?: GitHubProfile;
+}
+
 export default function Home() {
+
+  const [profile, setProfile] = useState<Profile>({})
+  const [languages, setLanguages] = useState<string[]>([])
+
+  useEffect(()=>{
+    axios.get(`https://ferrata.builtwithdark.com/open-profile/api/v1/profile/ferrata`)   //gotta make the username part dynamic `${username}`
```
Comment (ferrata) [inline src/pages/index.tsx]: If we could get a username from the URL path here, we could merge this PR to `main`. I think.

File: src/pages/languages.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import Head from "next/head";
+// import React from "react";
+
+
+
+///NOT SURE I NEED THIS COMPONENT!!!!!
```
Comment (ferrata) [inline src/pages/languages.tsx]: Probably delete this file for now? since we don't need it.

File: package.json [hunk 4]
```diff
@@ -13,11 +13,13 @@
     "@types/node": "18.11.18",
     "@types/react": "18.0.27",
     "@types/react-dom": "18.0.10",
+    "axios": "^1.3.3",
     "eslint": "8.32.0",
     "eslint-config-next": "13.1.4",
     "next": "13.1.4",
     "react": "18.2.0",
     "react-dom": "18.2.0",
+    "styled-components": "^5.3.6",
```
Comment (ferrata) [inline package.json:22]: Are we using `styled-components` in this PR?

File: src/pages/index.tsx [hunk 5]
```diff
@@ -1,10 +1,57 @@
...
+
+type GitHubProfile = {
+  languagesUsed: Map<string, number>;
+  profile?: GitHubUserProfile;
+  topFiveStarredRepos: GitHubRepository[];
+}
+
+type Profile = {
+  gitHub?: GitHubProfile;
+}
```
Comment (ferrata) [inline src/pages/index.tsx]: We should probably move these `types` to a separate file (or files)?

File: src/pages/languages.tsx (hunk 3 above)
Comment (Timbobeek) [reply src/pages/languages.tsx]: agree

File: src/pages/index.tsx (hunk 1 above)
Comment (Timbobeek) [reply src/pages/index.tsx]: yes, that looks better

File: src/pages/index.tsx (hunk 2 above)
Comment (Timbobeek) [reply src/pages/index.tsx]: yes

File: src/pages/index.tsx (hunk 5 above)
Comment (Timbobeek) [reply src/pages/index.tsx]: like a page just for types?

File: package.json (hunk 4 above)
Comment (Timbobeek) [reply package.json:22]: no, just checked

File: src/pages/index.tsx (hunk 5 above)
Comment (ferrata) [reply src/pages/index.tsx]: Just a separate file

**Pull request merged**